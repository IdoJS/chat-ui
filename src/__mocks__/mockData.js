const MOCK_USER_NAME = 'mock_user_name';
const MOCK_AVATAR_CLASS = 'mock_avatar_class';
const MOCK_TEXT = 'mock_text';
const MOCK_SOCKET_REQ = {text: MOCK_TEXT, avatar: MOCK_AVATAR_CLASS, userName: MOCK_USER_NAME, userId: '0'};
const MOCK_SELF_MESSAGE = {text: MOCK_TEXT, avatar: MOCK_AVATAR_CLASS, userName: MOCK_USER_NAME, timestamp: 0, userId: '0'};
const MOCK_OTHER_MESSAGE = {text: MOCK_TEXT, avatar: MOCK_AVATAR_CLASS, userName: 'other_user', timestamp: 0, userId: '1'};

const MOCK_CHAT_DATA_INIT = {
  loading: false,
  timestamp: Date.now(),
  chatArr: [],
  typingArr: [],
  timeout: false
};
export {
  MOCK_USER_NAME,
  MOCK_AVATAR_CLASS,
  MOCK_TEXT,
  MOCK_SOCKET_REQ,
  MOCK_CHAT_DATA_INIT,
  MOCK_SELF_MESSAGE,
  MOCK_OTHER_MESSAGE
}