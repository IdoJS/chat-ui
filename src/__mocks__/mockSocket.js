// modify the code https://medium.freecodecamp.org/testing-socket-io-client-app-using-jest-and-react-testing-library-9cae93c070a3

let EVENTS = {};
function emit(event, ...args) {
  EVENTS[event].forEach(func => func(...args));
}
const socket = () => {
  return {
    on(event, func) {
      if (EVENTS[event]) {
        return EVENTS[event].push(func);
      }
      EVENTS[event] = [func];
    },
    emit
  }
};
export const mockIo = {
  connect() {
    return socket;
  }
};

export default mockIo;