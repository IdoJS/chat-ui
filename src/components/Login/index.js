import React from 'react';
import PropTypes from 'prop-types';


class Login extends React.PureComponent {
  state = {
    avatarClass: '',
    userName: ''
  };

  constructor(props) {
    super(props);

    this.submit = this.submit.bind(this);
    this.onTextChange = this.onTextChange.bind(this);
    this.onAvatarChange = this.onAvatarChange.bind(this)
  }

  render() {
  const radios =  [...Array(5).keys()].map((index) => (<li key={index} className='ui image label'>
    <input type='radio' className='radio' name='avatar' id={`avatar_img0${index+1}`}/>
    <div className={`avatar avatar_img0${index+1}`}></div>
  </li>));

    return <form className='ui form' key={0}>
      <h1 className='ui large header'>Login</h1>
      <div class='field'>
        <label>Please enter userName</label>
        <div class='ui left icon input'>
          <input type='text'
                 className='ui input'
                 name='userName'
                 placeholder='Type name here...'
                 value={this.state.userName}
                 onChange={this.onTextChange}/>
          <i class='user circle icon'></i>
        </div>
      </div>

      <div class='field'>
        <label>Please choose avatar</label>
        <div className='ui list'>
          <ul className='form_radio_group' onChange={this.onAvatarChange.bind(this)}>
            {radios}
          </ul>
        </div>
      </div>

      <button className='ui primary button' onClick={this.submit}>Submit</button>
    </form>
  }

  onAvatarChange(ev) {
    this.setState({
      avatarClass: ev.target.id
    })
  }

  onTextChange(ev) {
    this.setState({
      [ev.target.name]: ev.target.value
    });
  }

  submit(ev) {
    ev.preventDefault();
    this.props.onUserCreate({
      userName: this.state.userName,
      avatarClass: this.state.avatarClass
    })
  }
};


Login.propTypes = {
  onUserCreate : PropTypes.func.isRequired
};

Login.defaultProps = {
  onUserCreate : () => {
    console.warn('Please implement onUserCreate');
  }
};

export default Login;