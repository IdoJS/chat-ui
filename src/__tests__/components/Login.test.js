import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';

import Login from '../../../src/components/Login';
import {MOCK_USER_NAME, MOCK_AVATAR_CLASS} from '../../__mocks__/mockData';

describe('Login component', () => {
  const ev = {
    preventDefault: () => {
    }
  };

  const evUserName = Object.assign({}, ev, {
    target: {
      value: MOCK_USER_NAME,
      name: 'userName'
    }
  });

  const evAvatar = Object.assign({}, ev, {
    target: {
      id: MOCK_AVATAR_CLASS,
      checked: true
    }
  });

  it('Should render the login view', (done) => {

    const wrapper = shallow(<Login onUserCreate={({userName, avatarClass}) => {
      expect(userName).toEqual(MOCK_USER_NAME);
      expect(avatarClass).toEqual(MOCK_AVATAR_CLASS);
      done()
    }}/>);

    const input = wrapper.find('input[type="text"]');
    const radioGroup = wrapper.find('div.avatar_radio_group');
    const radioButtons = radioGroup.find('input');
    const btn = wrapper.find('button');

    input.simulate('change', evUserName);

    expect(radioButtons).toHaveLength(5);

    radioGroup.simulate('change', evAvatar);

    btn.simulate('click', ev);
  });

  it('SnapShot rendering Login', () => {
    const wrapper = shallow(<Login/>);
    const tree = renderer.create(wrapper).toJSON();
    expect(tree).toMatchSnapshot();
  });

});