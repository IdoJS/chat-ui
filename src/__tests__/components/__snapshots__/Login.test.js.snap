// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component SnapShot rendering Login 1`] = `
<form
  className="form"
>
  <h1
    className="primary_title"
  >
    Login
  </h1>
  <div
    className="form_row"
  >
    <h2
      className="secondary_title"
    >
      Please enter userName
    </h2>
    <input
      className="primary_input"
      name="userName"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form_row"
  >
    <h2
      className="secondary_title"
    >
      Please choose avatar
    </h2>
    <div
      className="avatar_radio_group"
      onChange={[Function]}
    >
      <input
        className="avatar_radio"
        id="avatar_img01"
        name="avatar"
        type="radio"
      />
       
      <div
        className="avatar_img01"
      />
      <input
        className="avatar_radio"
        id="avatar_img02"
        name="avatar"
        type="radio"
      />
       
      <div
        className="avatar_img02"
      />
      <input
        className="avatar_radio"
        id="avatar_img03"
        name="avatar"
        type="radio"
      />
       
      <div
        className="avatar_img03"
      />
      <input
        className="avatar_radio"
        id="avatar_img04"
        name="avatar"
        type="radio"
      />
       
      <div
        className="avatar_img04"
      />
      <input
        className="avatar_radio"
        id="avatar_img05"
        name="avatar"
        type="radio"
      />
       
      <div
        className="avatar_img05"
      />
    </div>
  </div>
  <button
    className="primary_button big"
    onClick={[Function]}
  >
    Submit
  </button>
</form>
`;

exports[`Login component SnapShot rendering Login 1`] = `
<form
  className="form"
>
  <h1
    className="primary_title"
  >
    Login
  </h1>
  <div
    className="form_row"
  >
    <h2
      className="secondary_title"
    >
      Please enter userName
    </h2>
    <input
      className="primary_input"
      name="userName"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form_row"
  >
    <h2
      className="secondary_title"
    >
      Please choose avatar
    </h2>
    <div
      className="avatar_radio_group"
      onChange={[Function]}
    >
      <input
        className="avatar_radio"
        id="avatar_img01"
        name="avatar"
        type="radio"
      />
       
      <div
        className="avatar_img01"
      />
      <input
        className="avatar_radio"
        id="avatar_img02"
        name="avatar"
        type="radio"
      />
       
      <div
        className="avatar_img02"
      />
      <input
        className="avatar_radio"
        id="avatar_img03"
        name="avatar"
        type="radio"
      />
       
      <div
        className="avatar_img03"
      />
      <input
        className="avatar_radio"
        id="avatar_img04"
        name="avatar"
        type="radio"
      />
       
      <div
        className="avatar_img04"
      />
      <input
        className="avatar_radio"
        id="avatar_img05"
        name="avatar"
        type="radio"
      />
       
      <div
        className="avatar_img05"
      />
    </div>
  </div>
  <button
    className="primary_button big"
    onClick={[Function]}
  >
    Submit
  </button>
</form>
`;
